src/main/java/assignment/birds/OrderedDictionary.java

```diff
@@ -19,7 +15,7 @@ public class OrderedDictionary implements OrderedDictionaryADT {
     public Node findNode(DataKey k) throws DictionaryException {
         Node current = root;
         int comparison;
-        if (root.isEmpty()) {
+        if (isEmpty()) {
```

Comment:
with the deletion logic, it works out to be a bit less painful if we don't have to worry about adding some sort of placeholder node and instead just use null to represent an empty tree so I switched to that approach and had to update these accordingly